src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeRates.tsx

```diff
@@ -179,24 +177,45 @@ export const useGetTradeRates = () => {
   const { manualReceiveAddress, walletReceiveAddress } = useTradeReceiveAddress()
   const receiveAddress = manualReceiveAddress ?? walletReceiveAddress
 
-  useEffect(() => {
-    // Always invalidate tags when this effect runs - args have changed, and whether we want to fetch an actual quote
-    // or a "skipToken" no-op, we always want to ensure that the tags are invalidated before a new query is ran
-    // That effectively means we'll unsubscribe to queries, considering them stale
-    dispatch(swapperApi.util.invalidateTags(['TradeQuote']))
-
-    // Clear the slice before asynchronously generating the input and running the request.
-    // This is to ensure the initial state change is done synchronously to prevent race conditions
-    // and losing sync on loading state etc.
-    dispatch(tradeQuoteSlice.actions.clear())
-
-    // Early exit on any invalid state
-    if (bnOrZero(sellAmountCryptoPrecision).isZero()) {
-      setTradeRateInput(skipToken)
-      dispatch(tradeQuoteSlice.actions.setIsTradeQuoteRequestAborted(true))
-      return
-    }
-    ;(async () => {
+  const { data: tradeRateInput } = useQuery({
+    queryKey: [
+      'getTradeRateInput',
+      {
+        buyAsset,
+        sellAmountCryptoPrecision,
+        sellAsset,
+        userSlippageTolerancePercentageDecimal,
+        sellAssetUsdRate,
+        // TODO(gomes): all the below are what's causing trade input to refentially invalidate on wallet connect
+        // We will need to find a way to have our cake and eat it, by ensuring we get bip44 and other addy-related data (e.g voting power etc) to
+        // referentially invalidate, while ensuring the *initial* connection of a wallet when quotes were gotten without one, doesn't invalidate anything
+        sellAccountMetadata,
+        votingPower,
+        thorVotingPower,
+        receiveAccountMetadata,
+        sellAccountId,
+        isVotingPowerLoading,
+        isBuyAssetChainSupported,
+        receiveAddress,
+        isSnapshotApiQueriesRejected,
+      },
+    ],
+    queryFn: async () => {
+      // Always invalidate tags when this effect runs - args have changed, and whether we want to fetch an actual quote
+      // or a "skipToken" no-op, we always want to ensure that the tags are invalidated before a new query is ran
+      // That effectively means we'll unsubscribe to queries, considering them stale
+      dispatch(swapperApi.util.invalidateTags(['TradeQuote']))
+
+      // Clear the slice before asynchronously generating the input and running the request.
+      // This is to ensure the initial state change is done synchronously to prevent race conditions
+      // and losing sync on loading state etc.
+      dispatch(tradeQuoteSlice.actions.clear())
+
+      // Early exit on any invalid state
+      if (bnOrZero(sellAmountCryptoPrecision).isZero()) {
+        dispatch(tradeQuoteSlice.actions.setIsTradeQuoteRequestAborted(true))
+        return skipToken
```

Comment:
[suggestion]: Can do this if you like, since we do `tradeQuoteInput: tradeRateInput ?? skipToken,` below. Also its kinda funky having a query return a skipToken for another query 😅
```suggestion
        return
```

Reply:
Had the exact same thought refactoring it about this weirdness haha, agree implicit/undefined return is cleaner!

Reply:
https://github.com/shapeshift/web/pull/8256/commits/dcb904671c45e138ecb9708e0405e7c47411a530